#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "sphero_formation"

gen = ParameterGenerator()

gen.add("alignment_factor", double_t, 0, "Alignment factor", 1, 0, 2)
gen.add("cohesion_factor", double_t, 0, "Cohesion factor", 1, 0, 2)
gen.add("separation_factor", double_t, 0, "Separation factor", 1, 0, 2)
gen.add("max_speed", double_t, 0, "Maximum speed", 1, 0, 5)
gen.add("crowd_radius", double_t, 0, "Crowding avoid radius", 1, 0, 10)
gen.add("search_radius", double_t, 0, "Neighbour searching radius", 1.9, 0, 10)
gen.add("avoid_radius", double_t, 0, "Radius for obstacle avoidance", 1, 0, 3)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "Reynolds"))
